lib/Runtime/Library/ForInObjectEnumerator.cpp

```diff
@@ -182,7 +207,32 @@ namespace Js
                 if (TestAndSetEnumerated(propertyId) //checks if the property is already enumerated or not
                     && (attributes & PropertyEnumerable))
                 {
-                    return currentIndex;
+                    bool propertyShadowed = false;
```

Comment:
Umm... Is it worth it to of paying the price of looking up the property on all the intervening prototype chain object on every enumerable property, instead of just enumerating all the property in the chain without skipping to the first object with enumerable properties in the property chain so that the bit vector check is enough?